content/en/docs/tasks/run-application/horizontal-pod-autoscale.md

```diff
@@ -504,6 +504,33 @@ stops adjusting the target (and sets the `ScalingActive` Condition on itself
 to `false`) until you reactivate it by manually adjusting the target's desired
 replica count or HPA's minimum replica count.
 
+### Migrating Deployments and StatefulSets to horizontal autoscaling
+
+When an HPA is enabled, it is recommended that the value of `spec.replicas` of
+the Deployment and/or Statefulset object is removed.  Without doing so, anytime
+a change to that object is applied, example via `kubectl apply -f
+deployment.yaml`, will instruct Kubernetes to scale the current number of Pods
+to the desired value retained in the `spec.replicas` key. This may not be
+desired and could be troublesome when an HPA is active.
+
+Keep in mind that the removal of `spec.replicas` may incur a one time
+degradation of Pod counts as the default value of this key is 1 (reference
+[Deployment Replicas](/docs/concepts/workloads/controllers/deployment#replicas).
+Upon the update all Pods except 1 will begin their Termination procedures.  Any
+deployment application afterwards will behave as normal and respect a Rolling
+Update configuration as desired.  We can avoid this degradation, there are two
+methods of accomplishing, each depend on how you are modifying your deployments:
+
+- Manually edit the last applied API configuration following the following
```

Comment:
Similarly to what you have in the other point, I think we should be more explicit that these instructions are specifically for people who are using client-side apply to manage the resource in question.

Reply:
@KnVerey do people know though?  While the option `server-side` defaults to false, those that utilize CI, or helm, or <insert other fun method of deploying> may have no idea how that system utilizes this capability.  I welcome a change to this language, but I'm not entirely sure what to state here without throwing so much information, persons reading this become overwhelmed and only question what they need to accomplish.  Also keeping in mind that this flag is relatively new as well.

Reply:
Perhaps something like:

```suggestion
- When not using [Server-Side Apply](/docs/reference/using-api/server-side-apply/) (which applies to most usecases) Manually edit the last applied API configuration following the following
```

But I don't like how that looks/reads and now people need to figure out what that means before they can determine if they should follow these instructions :thinking:

Reply:
You could use [tabs](https://kubernetes.io/docs/contribute/style/hugo-shortcodes/#tabs) with SSA as default, to let readers see advice matched to the apply strategy they're using.

Do you think that could help @jtslear?

Reply:
I understand that some users might not know off hand which version they're using, but to avoid the degradation, they will need to find out to choose the appropriate mitigation. We could go into an explanation based on looking at the live object in question, but it might be easier for folks to look at their deploy pipeline than to read the metadata with `--show-managed-fields`. It seems to me that if you have `.metadata.annotations["kubectl.kubernetes.io/last-applied-configuration"]` and NOT `.metadata.managedFields`, then it's for sure client-side. But I can't think of a test that easy for server-side, notably because it keeps the old annotation in sync.

I like the tabs idea, but client-side should probably be the default since it's still the default in kubectl.

Reply:
@sftim I've added tabs, which really cleans this up, thanks for that pointer.

I'm still unsure of the best way to help people understand how to determine which deployment application method they are using without creating another large set of steps to be followed.  Having client side the default open tab, plus the comment indicating that this is the norm or default I think helps a bit.  Hopefully if persons are using SSA they'll know this as currently it requires people to know about said feature and explicitly utilize it.  (From my understanding of it thus far)

I welcome additional ideas on this, however.